File: WEB/src/app/features/users/services/admin-user.service.ts
import { inject, Injectable } from '@angular/core';
import {
	DriverRegistrationRequest,
	DriverRegistrationResponse,
} from '@features/users/models/drivers.model';
import { Observable } from 'rxjs';
import { HttpClient, HttpParams } from '@angular/common/http';
import { environment } from '@environments/environment';
import {
	UserDetailResponse,
	UserFilters,
	UserPageResponse,
} from '@features/users/models/user.model';

@Injectable()
export class AdminUserService {
	private http = inject(HttpClient);

	registerDriver(
		registrationRequest: DriverRegistrationRequest,
	): Observable<DriverRegistrationResponse> {
		return this.http.post<DriverRegistrationResponse>(
			`/api/${environment.apiVersion}/register/drivers`,
			registrationRequest,
		);
	}

	getUsers(
		page: number = 0,
		size: number = 10,
		sort?: string,
		direction: 'ASC' | 'DESC' = 'ASC',
		filters?: UserFilters,
	): Observable<UserPageResponse> {
		let params = new HttpParams().set('page', page.toString()).set('size', size.toString());

		if (sort) {
			params = params.set('sortBy', sort);
			params = params.set('sortDirection', direction);
		}

		// Add filters if provided
		if (filters) {
			Object.keys(filters).forEach((key) => {
				const value = filters[key as keyof UserFilters];
				if (value !== undefined && value !== null && value !== '') {
					params = params.set(key, value.toString());
				}
			});
		}

		return this.http.get<UserPageResponse>(`/api/${environment.apiVersion}/users`, { params });
	}

	getUserDetails(userId: number): Observable<UserDetailResponse> {
		return this.http.get<UserDetailResponse>(`/api/${environment.apiVersion}/users/${userId}`);
	}

	blockUser(userId: number, reason: string): Observable<void> {
		return this.http.patch<void>(`/api/${environment.apiVersion}/users/${userId}/block`, { reason });
	}

	unblockUser(userId: number): Observable<void> {
		return this.http.patch<void>(`/api/${environment.apiVersion}/users/${userId}/unblock`, {});
	}

	approveChangeRequest(requestId: number): Observable<void> {
		return this.http.post<void>(`/api/${environment.apiVersion}/profile-update-requests/${requestId}/approve`, {});
	}

	rejectChangeRequest(requestId: number): Observable<void> {
		return this.http.post<void>(`/api/${environment.apiVersion}/profile-update-requests/${requestId}/reject`, {});
	}
}

File: MOBILE/app/src/main/java/com/project/mobile/service/AdminUserService.java
package com.project.mobile.service;

import com.project.mobile.DTO.users.BlockUserRequest;
import com.project.mobile.DTO.users.DriverRegistrationRequest;
import com.project.mobile.DTO.users.DriverRegistrationResponse;
import com.project.mobile.DTO.users.UserDetailResponse;
import com.project.mobile.DTO.users.UserPageResponse;

import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.GET;
import retrofit2.http.PATCH;
import retrofit2.http.POST;
import retrofit2.http.Path;
import retrofit2.http.Query;

public interface AdminUserService {
    
    @GET("users")
    Call<UserPageResponse> getUsers(
        @Query("page") int page,
        @Query("size") int size,
        @Query("sortBy") String sortBy,
        @Query("sortDirection") String sortDirection,
        @Query("email") String email,
        @Query("firstName") String firstName,
        @Query("lastName") String lastName,
        @Query("role") String role,
        @Query("isBlocked") Boolean isBlocked,
        @Query("driverStatus") String driverStatus,
        @Query("hasPendingRequests") Boolean hasPendingRequests
    );

    @POST("register/drivers")
    Call<DriverRegistrationResponse> registerDriver(@Body DriverRegistrationRequest request);

    @GET("users/{userId}")
    Call<UserDetailResponse> getUserDetails(@Path("userId") long userId);

    @PATCH("users/{userId}/block")
    Call<Void> blockUser(@Path("userId") long userId, @Body BlockUserRequest request);

    @PATCH("users/{userId}/unblock")
    Call<Void> unblockUser(@Path("userId") long userId);

    @POST("profile-update-requests/{requestId}/approve")
    Call<Void> approveChangeRequest(@Path("requestId") long requestId);

    @POST("profile-update-requests/{requestId}/reject")
    Call<Void> rejectChangeRequest(@Path("requestId") long requestId);
}
